From 3bb0d5029f9f4cdebf0acd293f14f0251ec51d55 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:07 -0400
Subject: [tools] perf/trace: Add beautifier for clock_gettime's clk_id argument

Message-id: <1408462094-14194-139-git-send-email-jolsa@redhat.com>
Patchwork-id: 88092
O-Subject: [PATCH RHEL7.1 BZ1131394 138/325] perf trace: Add beautifier for clock_gettime's clk_id argument
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 4f8c1b74c5fdac35ee4480685d42030446724848
Author: David Ahern <dsahern@gmail.com>
Date: Sun Sep 22 19:45:00 2013 -0600

description
===========
Before:
0.030 ( 0.002 ms): 2571 clock_gettime(which_clock: 1, tp: 0x7f3b45729cd0 ) = 0

After:
0.030 ( 0.002 ms): 2571 clock_gettime(which_clock: MONOTONIC, tp: 0x7f3b45729cd0 ) = 0

v2: Update to use the STRARRAY option
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index adc7cd0..c993a8b 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -297,6 +297,12 @@ static DEFINE_STRARRAY(rlimit_resources);
 static const char *sighow[] = { "BLOCK", "UNBLOCK", "SETMASK", };
 static DEFINE_STRARRAY(sighow);
 
+static const char *clockid[] = {
+ "REALTIME", "MONOTONIC", "PROCESS_CPUTIME_ID", "THREAD_CPUTIME_ID",
+ "MONOTONIC_RAW", "REALTIME_COARSE", "MONOTONIC_COARSE",
+};
+static DEFINE_STRARRAY(clockid);
+
 static const char *socket_families[] = {
  "UNSPEC", "LOCAL", "INET", "AX25", "IPX", "APPLETALK", "NETROM",
  "BRIDGE", "ATMPVC", "X25", "INET6", "ROSE", "DECnet", "NETBEUI",
@@ -603,6 +609,7 @@ static struct syscall_fmt {
  { .name     = "arch_prctl", .errmsg = true, .alias = "prctl", },
  { .name     = "brk",     .hexret = true,
    .arg_scnprintf = { [0] = SCA_HEX, /* brk */ }, },
+ { .name     = "clock_gettime",  .errmsg = true, STRARRAY(0, clk_id, clockid), },
  { .name     = "connect",    .errmsg = true, },
  { .name     = "epoll_ctl",  .errmsg = true, STRARRAY(1, op, epoll_ctl_ops), },
  { .name     = "eventfd2",   .errmsg = true,
-- 
1.7.1